spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -336,25 +337,42 @@ function chanOpenTry(
   counterpartyChannelIdentifier: Identifier,
   version: string,
   counterpartyVersion: string,
-  proofInit: CommitmentProof,
+  proofInit: CommitmentProof | MultihopProof,
   proofHeight: Height): CapabilityKey {
     channelIdentifier = generateIdentifier()
 
     abortTransactionUnless(validateChannelIdentifier(portIdentifier, channelIdentifier))
-    abortTransactionUnless(connectionHops.length === 1) // for v1 of the IBC protocol
     abortTransactionUnless(authenticateCapability(portPath(portIdentifier), portCapability))
     connection = provableStore.get(connectionPath(connectionHops[0]))
     abortTransactionUnless(connection !== null)
     abortTransactionUnless(connection.state === OPEN)
+
+    let counterpartyHops: []
+    if (connectionHops.length > 1) {
+      counterpartyHops = getCounterPartyHops(proofInit)
+    } else {
+      counterpartyHops = [connection.counterpartyConnectionIdentifier]
+    }
+
     expected = ChannelEnd{INIT, order, portIdentifier,
-                          "", [connection.counterpartyConnectionIdentifier], counterpartyVersion}
-    abortTransactionUnless(connection.verifyChannelState(
-      proofHeight,
-      proofInit,
-      counterpartyPortIdentifier,
-      counterpartyChannelIdentifier,
-      expected
-    ))
+                          "", counterpartyHops, counterpartyVersion}
+
+    if (connectionHops.length > 1) {
+      consensusState = provableStore.get(consensusStatePath(connection.ClientId, proofHeight)
+      abortTransactionUnless(VerifyMultihopProof(
```

Comment:
We also need a similar check on ChanOpenAck.

Given that we will have all the connection ends in the MultiHopProof, we should check at least in these handshake messages

That the connection hops from A->B are the inverse of the connection hops from B->A

This is possible if we have the connection end structs between A and B